data-prepper-plugins/date-processor/src/main/java/com/amazon/dataprepper/plugins/processor/date/DateProcessor.java

```diff
@@ -12,37 +12,164 @@
 import com.amazon.dataprepper.model.processor.AbstractProcessor;
 import com.amazon.dataprepper.model.processor.Processor;
 import com.amazon.dataprepper.model.record.Record;
-import org.apache.commons.lang3.NotImplementedException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
 import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 @DataPrepperPlugin(name = "date", pluginType = Processor.class, pluginConfigurationType = DateProcessorConfig.class)
 public class DateProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
     private static final Logger LOG = LoggerFactory.getLogger(DateProcessor.class);
+    private static final ZoneId OUTPUT_TIMEZONE = ZoneId.systemDefault();
+    static final String OUTPUT_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX";
 
+    private String keyToParse;
+    private Locale sourceLocale;
+    private ZoneId sourceTimezone;
+    private List<DateTimeFormatter> dateTimeFormatters;
+    private IllegalArgumentException parsingException;
     private final DateProcessorConfig dateProcessorConfig;
 
     @DataPrepperPluginConstructor
-    protected DateProcessor(PluginMetrics pluginMetrics, final DateProcessorConfig dateProcessorConfig) throws NotImplementedException {
+    protected DateProcessor(PluginMetrics pluginMetrics, final DateProcessorConfig dateProcessorConfig) {
         super(pluginMetrics);
         this.dateProcessorConfig = dateProcessorConfig;
+
+        if (Boolean.TRUE.equals(dateProcessorConfig.getFromTimeReceived()) && dateProcessorConfig.getMatch() != null) {
+            throw new IllegalArgumentException("from_time_received and match are mutually exclusive options.");
+        }
+
+        else if (Boolean.FALSE.equals(dateProcessorConfig.getFromTimeReceived()) && dateProcessorConfig.getMatch() != null) {
+            sourceTimezone = buildZoneId(dateProcessorConfig.getTimezone());
+            if (dateProcessorConfig.getLocale() == null || dateProcessorConfig.getLocale().equalsIgnoreCase("ROOT")) {
+                sourceLocale = Locale.ROOT;
+            }
+            else {
+                sourceLocale = parseLocale(dateProcessorConfig.getLocale());
+            }
+
+            if (dateProcessorConfig.getMatch().size() != 1)
```

Comment:
Why are we limiting match to only 1 now? What if the user has a couple different types of logs with different timestamp formats going through the same pipeline? Do we want them to have to configure multiple `date` processors?

Also, I'm gonna emphasize again that all this validation can go in the config. It will make the `DateProcessor` constructor much cleaner and easier to follow.